examples/Autopoint/MultidimensionalGaussian_mixture_autopoint.cpp

```diff
@@ -0,0 +1,133 @@
+    // for truncated sampling and NULL for untruncated
+    HamiltonianMonteCarloWalk::Walk<Point, Hpolytope, RandomNumberGenerator, NegativeGradientFunctor, NegativeLogprobFunctor, Solver>
+        hmc(&P, x0, F, f, hmc_params);
+    int n_samples = 50000; // Half will be burned
+    int max_actual_draws = n_samples / 2;
+    unsigned int min_ess = 0;
+    MT samples;
+    samples.resize(dim, max_actual_draws);
+
+    for (int i = 0; i < n_samples - max_actual_draws; i++)
+    {
+        hmc.apply(rng, 3);
+    }
+    start = std::chrono::high_resolution_clock::now();
+    std::cerr << (long)std::chrono::duration_cast<std::chrono::microseconds>(start - stop).count();
+    for (int i = 0; i < max_actual_draws; i++)
+    {  std::cout << hmc.x.getCoefficients().transpose() << std::endl;
+        hmc.apply(rng, 3);
+        samples.col(i) = hmc.x.getCoefficients();
+    }
+    stop = std::chrono::high_resolution_clock::now();
+    long ETA = (long)std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count();
+    std::cerr << "total time taken " << ETA << std::endl;
+    std::cerr << std::endl;
+    print_diagnostics<NT, VT, MT>(samples, min_ess, std::cerr);
+    std::cerr << "min ess " << min_ess << "us" << std::endl;
+    std::cerr << "Average time per sample: " << ETA / max_actual_draws << "us" << std::endl;
+    std::cerr << "Average time per independent sample: " << ETA / min_ess << "us" << std::endl;
+    std::cerr << "Average number of reflections: " << (1.0 * hmc.solver->num_reflections) / hmc.solver->num_steps << std::endl;
+    std::cerr << "Step size (final): " << hmc.solver->eta << std::endl;
+    std::cerr << "Discard Ratio: " << hmc.discard_ratio << std::endl;
+    std::cerr << "Average Acceptance Probability: " << exp(hmc.average_acceptance_log_prob) << std::endl;
+    std::cerr << "PSRF: " << multivariate_psrf<NT, VT, MT>(samples) << std::endl;
+    std::cerr << std::endl;
+}
+using TT=double;
+typename autopoint<TT>::FT pdf_(const  autopoint<TT>& x,const Eigen::Matrix<TT,Eigen::Dynamic,Eigen::Dynamic>& data_)
+{
+
+
+    int d=x.getCoefficients().rows();
+    autopoint<TT> firstX=x.head(d/2);
+    autopoint<TT> secondX=x.tail(d/2);
+
+    typename autopoint<TT>::FT SUM=0;
+    auto I=Eigen::Matrix<TT,Eigen::Dynamic,Eigen::Dynamic>::Identity(d,d);
+    for(int i=0;i<data_.rows();i++)
+    {
+        Eigen::Matrix<TT,Eigen::Dynamic,1> datavector=data_.row(i).transpose();
+        autopoint<TT> data_autopoint= autopoint<TT>(datavector);
+        autopoint<TT> x_mu_first= data_autopoint - firstX;
+        autopoint<TT> x_mu_second= data_autopoint -secondX;
+        SUM +=(-log(exp(-0.5*x_mu_first.pow(2).sum()) + exp(-0.5*x_mu_second.pow(2).sum() ) ));
+    }
+    return SUM;
+
+}
+template <> std::function<typename autopoint<TT>::FT(const autopoint<TT>&,const Eigen::Matrix<TT,Eigen::Dynamic,Eigen::Dynamic>&)>  AutoDiffFunctor::FunctionFunctor_internal<TT>::pdf=pdf_;
```

Comment:
this line is too long, could you please split it ?

Reply:
used typedef and reduced the length slightly